src/pages/Embeds/Plio.vue

```diff
@@ -316,6 +398,83 @@ export default {
   },
   methods: {
     ...mapActions("auth", ["setAccessToken", "setActiveWorkspace"]),
+    /**
+     * Update the number of correct answers
+     * @param  {Number} itemIndex -  The index of the item to be considered for the calculation
+     * @param  {String, Number, Object} userAnswer - User's answer to that item
+     */
+    updateNumCorrect(itemIndex, userAnswer) {
+      if (this.isItemMCQ(itemIndex)) {
+        const correctAnswer = this.items[itemIndex].details.correct_answer;
+        if (!isNaN(userAnswer) && userAnswer == correctAnswer) {
+          this.numCorrect += 1;
+        }
+      } else if (
+        this.isItemSubjective(itemIndex) &&
+        userAnswer != null &&
+        userAnswer.trim() != ""
+      ) {
+        this.numCorrect += 1;
+      }
+      return;
+    },
+    /**
+     * Update the number of wrong answers
+     * @param  {Number} itemIndex -  The index of the item to be considered for the calculation
+     * @param  {String, Number, Object} userAnswer - User's answer to that item
+     */
+    updateNumWrong(itemIndex, userAnswer) {
+      if (this.isItemMCQ(itemIndex)) {
+        const correctAnswer = this.items[itemIndex].details.correct_answer;
+        if (!isNaN(userAnswer) && userAnswer != correctAnswer) {
+          this.numWrong += 1;
+        }
+      }
+      return;
+    },
+    /**
+     * Update the number of skipped answers
+     * @param  {Number} itemIndex -  The index of the item to be considered for the calculation
+     * @param  {String, Number, Object} userAnswer - User's answer to that item
+     */
+    updateNumSkipped(itemIndex, userAnswer) {
+      if (this.isItemMCQ(itemIndex) && isNaN(userAnswer)) {
+        this.numSkipped += 1;
+      } else if (this.isItemSubjective(itemIndex)) {
+        if (userAnswer == null || userAnswer.trim() == "") this.numSkipped += 1;
```

Comment:
this condition can be wrapped under: "isSubjectiveQuestionAnswered" - and can be used in multiple places